### PR TITLE
fix: Unread message count shows wrong at the top bar. #2097

### LINKED ISSUE #2097: Unread message count shows wrong at the top bar.
**Describe the bug**
Unread message count shows wrong at the top bar because it also counting unread messages from hidden threads.

**To Reproduce**
1. Send a message to Jessica
2. Login as Jessica then go the the messages page.
3. Then click on the (x) icon for the newly appeared thread without viewing the messages.
4. Now you should see the unread message count for the hidden threads and there is no way to remove the unread message count from the message notification area.

**Expected behavior**
There should not be unread message count for hidden threads

**Screenshots**
[https://prnt.sc/xxlg96](https://prnt.sc/xxlg96)

**Support ticket links**
[https://secure.helpscout.net/conversation/1406883779/121947?folderId=4178440](https://secure.helpscout.net/conversation/1406883779/121947?folderId=4178440)

### PR DESCRIPTION
**Pull request by milanhirapra**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2097 .

### How to test the changes in this Pull Request:

1. Send a message to Jessica
2. Login as Jessica then go to the messages page.
3. Then click on the (x) icon for the newly appeared thread without viewing the messages.
4. Now you can see the **unread message count** and unread message count from the **message notification** for the hidden threads are marked as read.

### Proof Screenshots or Video
https://vimeo.com/591572578/3bec52652e

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dipskakadiya — APPROVED** [review]
@milanhirapra Approved. Ready for testing

**Comment by harshalQA** [comment]
Fixed: Retested on site url: https://bb-app-platform.flywheelsites.com/wp-admin

Video:
https://www.loom.com/share/2718451c45ec4c9ca484155744953bae

File: src/bp-templates/bp-nouveau/includes/messages/ajax.php [hunk 1]
```diff
@@ -2599,7 +2599,32 @@ function bp_nouveau_ajax_hide_thread() {
...
+
+			if ( ! empty( $unread_message_ids ) ) {
+
+				// Mark each notification for each PM message as read.
+				foreach ( $unread_message_ids as $message_id ) {
+					bp_notifications_mark_notifications_by_item_id( bp_loggedin_user_id(), (int) $message_id, buddypress()->messages->id, 'new_message' );
+				}
+
+			}
+		}
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/messages/ajax.php:2627]: @milanhirapra Please move this code outside the loop.

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

File: src/bp-templates/bp-nouveau/includes/messages/ajax.php (hunk 1 above)
Comment (milanhirapra) [reply src/bp-templates/bp-nouveau/includes/messages/ajax.php]: @KartikSuthar It's done! Please check and review.

**Comment by JetalDobariya** [comment]
Hello @milanhirapra @KartikSuthar 

I did verify this PR in qa-approved branch, now Unread message count working fine after read messages.
Unread Messages Count --> https://www.loom.com/i/478f1ae673604e5c85959ea5333a691c
After Read Messaged Count --> https://www.loom.com/i/7b48927e41c24ee8a5854ea75e8e31e0

Thanks!
CC : @harshalQA

**Pull request merged**